### PR TITLE
MM-33215 Add documentation for working with GitLab Omnibus

### PR DESCRIPTION
**Pull request by hmhealey**
A full guide to what I do to test and submit new versions of MM with GItLab Omnibus so that that can still be done if I get hit by a bus randomly.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33215

### CODE REVIEW

File: site/content/internal/gitlab-omnibus.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+1. Ensure our fork's master branch is up to date with upstream.
+
+    ```bash
+    git checkout master
+    git fetch upstream
+    git pull upstream master
+    git push
+    ```
+       
+1. Create a branch for the new version of Mattermost.
```
Comment (lieut-data) [inline site/content/internal/gitlab-omnibus.md:47]: nit: I know this is handled at render time, but can we increment the list here for easy "plaintext" reading too?

**Review by lieut-data — APPROVED** [review]
This is terrific work, @hmhealey! Though I'm afraid we'll still have to revoke your bus pass just to .. you know, make sure :)

File: site/content/internal/gitlab-omnibus.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    external_url "http://gitlab.dev.mm"
+    ```
+
+1. Find and uncomment the line which sets `mattermost_external_url` and set it to Mattermost's external URL.
+
+    ```ruby
+    mattermost_external_url "http://mattermost.dev.mm"
+    ```
+
+1. Find and uncommonet the line which sets `mattermost['enable']` and set it to `true`.
```
Comment (metanerd) [inline site/content/internal/gitlab-omnibus.md:157]:
```suggestion
1. Find and uncomment the line which sets `mattermost['enable']` and set it to `true`.
```

File: site/content/internal/gitlab-omnibus.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+    1. In Mattermost, go to Integrations > Incoming Webhooks and add an incoming webhook.
+    
+    1. Copy the URL of the webhook and return to Mattermost.
+
+    1. Go to GitLab. From the sidebar on the left, go to Settings > Integrations.
+
+    1. Scroll down and select "Mattermost notifications".
+
+    1. Fill in the "Webhook" field with the webhool URL you previously copied and then click "Save changes"
```
Comment (metanerd) [inline site/content/internal/gitlab-omnibus.md:229]:
```suggestion
    1. Fill in the "Webhook" field with the webhook URL you previously copied and then click "Save changes"
```

**Review by metanerd — APPROVED** [review]
Thank you for this detailed description @hmhealey ! I think a lot of these steps could be automated within the new flow: https://mattermost.atlassian.net/browse/DOPS-351 . I created a ticket for that and we will prioritize it accordingly.

**Comment by hmhealey** [comment]
Yeah, given it's all done in a Docker container, it definitely seemed prime for automation. It does require a few hours to run on a pretty beefy VM (I can't remember if mine is a medium or large, but it has 8 gigs of ram), so I don't imagine CircleCI would want to run it for us, but I guess we could even use our GitLab server for it.

File: site/content/internal/gitlab-omnibus.md [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+    The type will usually be "8. Other". You can also go back afterwards and add the author and MR number if you'd like.
+   
+6. Commit the changes to `config/software/mattermost.rb`, `doc/gitlab-mattermost/README.md`, and the newly created file in `changelogs/unreleased`. The commit message must also a second line containing the type of change made similar to the one listed in the changelog file.
+
+    ```
+    Update Mattermost to X.Y
+    Changelog: other
+    ```
+
+You can now test the build and submit a MR upstream. For an example of how the branch should look after that, see [here](https://gitlab.com/gitlab-org/omnibus-gitlab/-/merge_requests/5068).
```
Comment (justinegeffen) [inline site/content/internal/gitlab-omnibus.md:72]:
```suggestion
You can now test the build and submit an MR upstream. For an example of how the branch should look after that, see [here](https://gitlab.com/gitlab-org/omnibus-gitlab/-/merge_requests/5068).
```

File: site/content/internal/gitlab-omnibus.md [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+
+### Building GitLab Omnibus
+
+As mentioned above, GitLab Omnibus is built in a Docker container containing all of the needed dependencies. To test it, the generated .deb package needs to be copied off of the Docker container and installed locally on the test server. Details on how to do that follow.
```
Comment (justinegeffen) [inline site/content/internal/gitlab-omnibus.md:76]:
```suggestion
As mentioned above, GitLab Omnibus is built in a Docker container containing all of the needed dependencies. To test it, the generated `.deb` package needs to be copied off of the Docker container and installed locally on the test server. Details on how to do that follow.
```

**Review by justinegeffen — APPROVED** [review]
Thanks, @hmhealey and apologies for the delay. The majority of my feedback is cosmetic - formatting and a few rewording suggestions. Those aside, this reads really well and I'm going to approve it ahead of your review/commit of the suggestions so as to expedite the process. :)

**Pull request merged**